crates/apub/src/activities/block/undo_block_user.rs

```diff
@@ -99,23 +100,33 @@ impl ActivityHandler for UndoBlockUser {
     let expires = self.object.end_time;
     let mod_person = self.actor.dereference(context).await?;
     let blocked_person = self.object.object.dereference(context).await?;
+    let pool = &mut context.pool();
     match self.object.target.dereference(context).await? {
-      SiteOrCommunity::Site(_site) => {
+      SiteOrCommunity::Site(site) => {
         verify_is_public(&self.to, &self.cc)?;
-        let blocked_person = Person::update(
-          &mut context.pool(),
-          blocked_person.id,
-          &PersonUpdateForm {
-            banned: Some(false),
-            ban_expires: Some(expires),
-            ..Default::default()
-          },
-        )
-        .await?;
-
-        if self.restore_data.unwrap_or(false) {
-          remove_or_restore_user_data(mod_person.id, blocked_person.id, false, &None, context)
-            .await?;
+        if blocked_person.instance_id == site.instance_id {
+          if self.restore_data.unwrap_or(false) {
+            remove_or_restore_user_data(mod_person.id, blocked_person.id, false, &None, context)
+              .await?;
+          }
+          // user unbanned from home instance, write directly to person table
+          InstanceActions::unban(
```

Comment:
Might as well move this out of the block, just like `block_user.rs` already does.